File: benchmarks/primes/primes.py
def main():
    n = 3000000
    sieve = [False] * (n + 1)
    count = 0
    for i in range(2, n + 1):
        if not sieve[i]:
            count += 1
            for j in range(i + i, n + 1, i):
                sieve[j] = True
    print("Primes:", count)

main()

File: benchmarks/primes/primes.java
class primes {
    public static void main(String[] args) {
        int n = 3000000;
        boolean[] sieve = new boolean[n + 1];
        int count = 0;
        for (int i = 2; i <= n; i++) {
            if (!sieve[i]) {
                count++;
                for (int j = i + i; j <= n; j += i)
                    sieve[j] = true;
            }
        }
        System.out.println("Primes: " + count);
    }
}
